### PR TITLE
add upload to s3

### PR DESCRIPTION
**Pull request by xspronken**
Adding the job that uploads the benchmarks to the S3 server to the workflow.

### CODE REVIEW

**Comment by xspronken** [comment]
@hodgestar have the secrets you set up to connect to the s3 bucket been removed?

**Comment by hodgestar** [comment]
The secrets are till there. Perhaps they aren't provided because this PR isn't from an fork?

**Review by hodgestar — APPROVED** [review]

**Review by hodgestar — CHANGES_REQUESTED** [review]
We should make sure this only runs on the main branch and only on this repo.

**Review by hodgestar — COMMENTED** [review]
From discussion: Let's
- copy run_benchmarks.yml to nightly_benchmarks.yml
- rename run_benchmarks.yml to test.yml
- have upload_S3.yml run only after nightly_benchmarks.yml
- change nightly benchmarks to run nightly.

File: .github/workflows/run_benchmarks.yml [hunk 1]
```diff
@@ -1,9 +1,10 @@
-name: Benchmarks
-
+name: Nightly Benchmarks
```
Comment (hodgestar) [inline .github/workflows/run_benchmarks.yml:1]: The file still needs to be renamed to "nightly_benchmarks.yml".

File: .github/workflows/upload_S3.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: Upload S3
```
Comment (hodgestar) [inline .github/workflows/upload_S3.yml:1]: I'd like to suggest that we just add this as an additional job to the nightly_benchmarks.yml. It simplifies things a bit, and, in particular, means we can use the default github artifact action.

So:
```
jobs:
    benchmarks:
       ...
    upload_to_s3:
       depends-on: ["benchmarks"]
       ...
```

File: .github/workflows/run_benchmarks.yml (hunk 1 above)
Comment (xspronken) [reply .github/workflows/run_benchmarks.yml]: Fixed in [faa9088](https://github.com/qutip/qutip-benchmark/pull/10/commits/faa9088d9dc9e0ebcc6e8500f505278e41ee1819).

File: .github/workflows/upload_S3.yml (hunk 2 above)
Comment (xspronken) [reply .github/workflows/upload_S3.yml]: I agree, I changed it in [fdf4eac](https://github.com/qutip/qutip-benchmark/pull/10/commits/fdf4eac49582c96e9f9ac0d4b9cef8dc7d296fb1).

File: .github/workflows/nightly_benchmarks.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
+name: Nightly Benchmarks
+
+on:
+  schedule:
+    # Run every day at midnight
+    - cron: "0 0 * * *"
+
```
Comment (hodgestar) [inline .github/workflows/nightly_benchmarks.yml:7]:
```suggestion
  workflow_dispatch

```
Let's also add `workflow_dispatch` then we can push a button in the GitHub interface to run the action.

File: .github/workflows/nightly_benchmarks.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        run: |
+          python benchmarks/benchmarks.py -m "nightly" -v 
+        
+      - name: Create artefact containing benchmarks
+        uses: actions/upload-artifact@v2
+        with: 
+          name: nightly_benchmarks
+          path: .benchmarks
+
+
```
Comment (hodgestar) [inline .github/workflows/nightly_benchmarks.yml:76]:
```suggestion
```

**Review by hodgestar — COMMENTED** [review]
Other than the last two small suggestions, looks good to me.

**Review by hodgestar — APPROVED** [review]

**Comment by hodgestar** [comment]
@xspronken Happy for you to merge this when you're ready. Don't forget to write a nice merge commit message when you do.

**Pull request merged**